core/spec/lib/spree/core/controller_helpers/store_spec.rb

```diff
@@ -45,6 +45,45 @@ class FakesController < ApplicationController
     end
   end
 
+  describe '#ensure_current_store' do
+    let!(:store) { create :store, default: true }
+    let!(:store_2) { create :store }
+
+    context 'on an object that accepts multiple stores' do
+      before { allow(controller).to receive(:current_store).and_return(store) }
+
+      object = Spree::Product.new
+
+      it 'object has only the current store set' do
+        controller.ensure_current_store(object)
+        expect(object.stores).to contain_exactly(store)
+        expect(object.stores).not_to contain_exactly(store_2)
+      end
+    end
+
+    context 'on a object that accepts a single store' do
```

Comment:
@MatthewKennedy let's also add a test for objects that aren't tied to a store :)

Reply:
I was just thinking about this. I'll get this sorted.